hazelcast/src/test/java/com/hazelcast/map/AsyncTest.java

```diff
@@ -184,4 +188,23 @@ public void testRemoveAsyncWithNonExistentKey() throws Exception {
         Future<String> f1 = map.removeAsync(key).toCompletableFuture();
         assertNull(f1.get());
     }
+
+    @Test
+    public void testValueAsync() throws Exception {
+        IMap<String, String> map = instance.getMap(randomString());
+        map.put(key, value1);
+        CompletableFuture<Collection<String>> futureString = map.valuesAsync().toCompletableFuture();
+        assertEquals(futureString.get().size(), 1);
+        assertTrue(futureString.get().iterator().hasNext());
+    }
+
+    @Test
+    public void testValueAsyncWithPredicate() throws Exception {
+        IMap<String, String> map = instance.getMap(randomString());
+        map.put(key, value1);
+        CompletableFuture<Collection<String>> futureString = map.valuesAsync(Predicates.alwaysFalse()).toCompletableFuture();
+        assertEquals(futureString.get().size(), 0);
+        assertFalse(futureString.get().iterator().hasNext());
```

Comment:
I would test with a different predicate. What about putting 100 entries and selecting first 20 etc

Reply:
changed test with different Predicate, please review again!